server/zanata-war/src/main/java/org/zanata/adapter/FileFormatAdapter.java

```diff
@@ -139,4 +143,36 @@ default String generateTranslationFilename(@Nonnull HDocument document, @Nonnull
         return FilenameUtils.removeExtension(document
                 .getName()) + "." + transExt;
     }
+
+    /**
+     * Read the data from the given file URI to a BufferedInputStream
+     *
+     * @param fileUri
+     * @return BufferedInputStream of file data
+     * @throws FileFormatAdapterException
+     * @throws IllegalArgumentException
+     */
+    default BufferedInputStream readStream(URI fileUri)
```

Comment:
This method doesn't belong in this interface. We should put it in a package-private utility function, because we shouldn't even be passing around file URIs, really.